src/stage1/exe_copy.rs

```diff
@@ -108,66 +92,16 @@ impl ExeCopy {
         }
     }
 
-    fn get_libs(&self, file: &str, ldd_path: &str, found: &mut HashSet<String>) -> Result<()> {
+    fn get_libs(&self, file: &str, found: &mut HashSet<String>) -> Result<()> {
         trace!("get_libs: entered with '{}'", file);
-        let ldd_res = match call(ldd_path, &[file], true) {
-            Ok(cmd_res) => {
-                if cmd_res.status.success() {
-                    cmd_res.stdout
-                } else if cmd_res.stderr.contains("not a dynamic executable")
-                    || cmd_res.stdout.contains("not a dynamic executable")
-                {
-                    return Ok(());
-                } else {
-                    return Err(Error::with_context(
-                        ErrorKind::ExecProcess,
-                        &format!(
-                            "Failed to retrieve dynamic libs for '{}', error: {}",
-                            file, cmd_res.stderr
-                        ),
-                    ));
-                }
-            }
-            Err(why) => {
-                return Err(Error::with_all(
-                    ErrorKind::ExecProcess,
-                    &format!("2failed to retrieve dynamic libs for '{}'", file),
-                    Box::new(why),
-                ));
-            }
-        };
-
-        lazy_static! {
-            static ref LIB_REGEX: Regex = Regex::new(
-                r#"^\s*(\S+)\s+(=>\s+(\S+)\s+)?(\(0x[0-9,a-f,A-F]+\))|statically linked$"#
-            )
-            .unwrap();
-        }
-
-        for lib_str in ldd_res.lines() {
-            if let Some(captures) = LIB_REGEX.captures(lib_str) {
-                if let Some(lib_name) = captures.get(1) {
-                    let lib_name = lib_name.as_str();
-                    if let Some(lib_path) = captures.get(3) {
-                        let lib_path = lib_path.as_str();
-                        if !self.libraries.contains(lib_path) {
-                            found.insert(lib_path.to_owned());
-                        }
-                    } else {
-                        let lib_path = PathBuf::from(lib_name);
-                        if !self.libraries.contains(lib_name) {
-                            if lib_path.is_absolute() && lib_path.exists() {
-                                found.insert(lib_name.to_owned());
-                            } else {
-                                debug!("get_libs: no path for {}", lib_name);
-                            }
-                        }
-                    }
-                } else {
-                    debug!("get_libs: lib is statically linked: '{}'", file);
-                }
-            } else {
-                warn!("get_libs: no match for {}", lib_str);
+        let analyzer = DependencyAnalyzer::new(PathBuf::from("/"));
+        let dependencies = analyzer.analyze(file).unwrap();
```

Comment:
Better to handle the error (even if only propagating up the call stack) for nicer error reporting (because `unwrap()` can panic).

Reply:
Thanks, looks good! :ok_hand: